tests/rptest/tests/partition_movement_test.py

```diff
@@ -43,17 +45,68 @@ class PartitionMovementTest(EndToEndTest):
     """
     def __init__(self, ctx, *args, **kwargs):
-        super(PartitionMovementTest, self).__init__(
-            ctx,
-            *args,
-            extra_rp_conf={
-                # Disable leader balancer, as this test is doing its own
-                # partition movement and the balancer would interfere
-                'enable_leader_balancer': False
-            },
-            **kwargs)
+        self._extra_rp_conf = {
+            # Disable leader balancer, as this test is doing its own
+            # partition movement and the balancer would interfere
+            'enable_leader_balancer': False
+        }
+
+        self.s3_client = None
+        self.s3_bucket_name = None
+        self.s3_bucket_name = f"panda-bucket-{uuid.uuid1()}"
+        self.s3_host_name = "minio-s3"
+        self.s3_bucket_name = f"panda-bucket-{uuid.uuid1()}"
+        self.s3_access_key = "panda-user"
+        self.s3_secret_key = "panda-secret"
+        self.s3_region = "panda-region"
+        self.s3_topic_name = "panda-topic"
+
+        if ctx.function_name == "test_shadow_indexing":
+            self._extra_rp_conf.update(
+                dict(
+                    cloud_storage_enabled=True,
+                    cloud_storage_access_key=self.s3_access_key,
+                    cloud_storage_secret_key=self.s3_secret_key,
+                    cloud_storage_region=self.s3_region,
+                    cloud_storage_bucket=self.s3_bucket_name,
+                    cloud_storage_disable_tls=True,
+                    cloud_storage_api_endpoint=self.s3_host_name,
+                    cloud_storage_api_endpoint_port=9000,
+                    cloud_storage_reconciliation_interval_ms=500,
+                    cloud_storage_max_connections=5,
+                    cloud_storage_enable_remote_read=True,
+                    cloud_storage_enable_remote_write=True,
+                    #log_compaction_interval_ms=10000,
+                    log_segment_size=10240,
+                    delete_retention_ms=1000,
+                ))
+
+        super(PartitionMovementTest,
+              self).__init__(ctx,
+                             *args,
+                             extra_rp_conf=self._extra_rp_conf,
+                             **kwargs)
         self._ctx = ctx
 
+        if ctx.function_name == "test_shadow_indexing":
+            self.s3_client = S3Client(
+                region='panda-region',
+                access_key=u"panda-user",
+                secret_key=u"panda-secret",
+                endpoint=f'http://{self.s3_host_name}:9000',
+                logger=self.logger)
+
+    def setUp(self):
+        if self.s3_client is not None:
+            self.s3_client.empty_bucket(self.s3_bucket_name)
+            self.s3_client.create_bucket(self.s3_bucket_name)
+        super().setUp()
+
+    def tearDown(self):
+        if self.s3_client is not None:
+            self.s3_client.empty_bucket(self.s3_bucket_name)
+        super().tearDown()
```

Comment:
does this fire? I was under the impression that ducktape didn't call any tear down methods, but it's been a long time since I remember looking into it.

Reply:
I'm pretty sure it is. At least a while ago I was debugging this since it caused issues in `archival_test.py`. The test was able pick up objects from previous test without this.